config/jobs/cert-manager/cert-manager/cert-manager-presubmits.yaml

```diff
@@ -660,6 +684,12 @@ presubmits:
           readOnly: true
         - mountPath: /sys/fs/cgroup
           name: cgroup
+        - mountPath: /root/.cache/go-build
+          name: gocache
+        - mountPath: /home/prow/go/pkg/mod
+          name: gopkgmod
+        - mountPath: /home/prow/go/src/github.com/cert-manager/cert-manager/bin/downloaded
```

Comment:
How does cache invalidation work for this if we say, bump a version of a binary dependency?

& how do we handle this between branches? (e.g. `release-1.9` and `master` may use different versions)

Reply:
Regarding cache invalidation, all the tools are named after their version number in `bin/downloaded`:

```
$ ll bin/tools
lrwxrwxrwx 31 mvalais 24 Mar 19:23 crane -> ../downloaded/tools/crane@0.8.0
lrwxrwxrwx 36 mvalais 24 Mar 19:24 etcd -> ../downloaded/tools/etcd-linux-amd64
lrwxrwxrwx 35 mvalais 24 Mar 19:24 gotestsum -> ../downloaded/tools/gotestsum@1.7.0
lrwxrwxrwx 43 mvalais 24 Mar 19:23 helm -> ../downloaded/tools/helm-v3.8.0-linux-amd64
lrwxrwxrwx 43 mvalais 24 Mar 19:23 kind -> ../downloaded/tools/kind_0.11.1_linux_amd64
lrwxrwxrwx 46 mvalais 24 Mar 19:24 kube-apiserver -> ../downloaded/tools/kube-apiserver-linux-amd64
lrwxrwxrwx 46 mvalais 24 Mar 19:23 kubectl -> ../downloaded/tools/kubectl_1.22.1_linux_amd64
lrwxrwxrwx 41 mvalais 24 Mar 19:23 yq -> ../downloaded/tools/yq_4.11.2_linux_amd64
```

Let's imagine we change the version of `kind` to 0.12.0 in the makefile. When the target `bin/tools/bind` will be evaluated, the symlink will get overridden to point to the new version.

The same mechanism works when switching branches: the symlink gets updated by make.

Reply:
Would that same apply for two jobs running concurrently on the same node?

Reply:
Yes, no problem. Each job will get its own `bin/tools`, and make will do the right symlinking. The only "shared" thing between the two jobs would be `bin/downloaded`.

Reply:
I realized that the tool symlinks weren't properly updated when going back and forth twice. I fixed this issue at https://github.com/cert-manager/cert-manager/pull/4988.